Modix.Data/Repositories/QueryPageBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.EntityFrameworkCore;
+
+namespace Modix.Data.Repositories
+{
+    public delegate IQueryable<T> QueryClause<T>(IQueryable<T> query);
+
+    public delegate IOrderedQueryable<T> OrderByClause<T>(IQueryable<T> query);
+
+    public class QueryPageBuilder<T>
+    {
+        public IQueryable<T> Query { get; set; }
+
+        public QueryClause<T> WhereClause { get; set; }
+
+        public OrderByClause<T> OrderByClause { get; set; }
+
+        public int? MaxPageSize { get; set; }
+
+        public async Task<QueryPage<T>> BuildAsync(PagingCriteria pagingCriteria)
+        {
+            if (Query == null)
+                return new QueryPage<T>()
```

Comment:
Would be nice to have a body to the `if` statement here, because it's multiple lines nesting looks a bit odd

Reply:
Personal preferences are personal. See above comment regarding style conventions.